great_expectations/rule_based_profiler/config/base.py

```diff
@@ -408,32 +415,58 @@ def resolve_config_using_acceptable_arguments(
     ) -> "RuleBasedProfilerConfig":  # noqa: F821
         runtime_config: RuleBasedProfilerConfig = profiler.config
 
-        runtime_variables: Optional[
-            ParameterContainer
-        ] = profiler.reconcile_profiler_variables(variables)
-        runtime_variables_configs: dict = (
-            runtime_variables.to_dict()["parameter_nodes"]["variables"]["variables"]
-            or {}
+        runtime_variables: dict = profiler.reconcile_profiler_variables_as_dict(
+            variables=variables
         )
-
-        effective_rules: List["Rule"] = profiler.reconcile_profiler_rules(  # noqa: F821
-            rules=rules
+        variables_container: ParameterContainer = (
+            build_parameter_container_for_variables(variables_configs=runtime_variables)
         )
 
-        rule: "Rule"  # noqa: F821
-        runtime_rules: Dict[str, dict] = {
-            rule.name: rule.to_json_dict() for rule in effective_rules
-        }
+        effective_rules: Dict[
+            str, "Rule"  # noqa: F821
+        ] = profiler.reconcile_profiler_rules_as_dict(rules=rules)
+        runtime_rules: Dict[str, dict] = {}
+        for name, rule in effective_rules.items():
+            rule_with_substituted_vars: dict = (
+                RuleBasedProfilerConfig._interpolate_variables_in_config(
+                    rule=rule, variables_container=variables_container
+                )
+            )
+            runtime_rules[name] = rule_with_substituted_vars
 
         return cls(
             class_name=profiler.__class__.__name__,
             module_name=profiler.__class__.__module__,
             name=runtime_config.name,
             config_version=runtime_config.config_version,
-            variables=runtime_variables_configs,
+            variables=runtime_variables,
             rules=runtime_rules,
         )
 
+    @staticmethod
+    def _interpolate_variables_in_config(
```

Comment:
@cdkini A little docstring for people not on this project might help for how `_traverse_and_interpolate` works and why it is there.  I also feel that the name "interpolate" does not fit (to me "expand" or "substitute" would fit better).  Thanks.

Reply:
I'm basing "interpolate" on how it's denoted in JavaScript when using template literals but I'm happy to change it to substitute.